### DISCUSSION TITLE
Proxmox replication not detected with 'exclude-unchanged 1'

### DISCUSSION THREAD

**Discussion by rholighaus** [psy0rz/zfs_autobackup#255] [Q&A] [answered]
I am testing zfs-autobackup on a Proxmox cluster that uses replication, in many cases hourly or more often. I want to replace znapzend as it still doesn't support resuming an abandoned offsite replication, which happens when our provider nightly cuts the VDSL link we exclusively use for remote backup.

For test purposes, I have set autobackup:test on two datasets, one of them is local (subvol-130-disk-0), the other one is a replication (subvol-129-disk-0). For testing, I am using this script: 

```
#!/bin/bash

DESTINATION=<hostname of the destination>

# backup non samba VMs/CTs
# (different snapshot timestamp format)

TAG=test
# change that later to 'offsite'

TARGET=rpool/carrier
# remove that later after tests

FORMAT='%Y-%m-%d-%H%M%S'

zfs-autobackup \
	--keep-source=1h1d,1d1w,1w1m \
	--snapshot-format $FORMAT \
	--ssh-target $DESTINATION \
	--clear-mountpoint \
	--clear-refreservation \
	--ignore-transfer-errors \
	--strip-path 2 \
	--verbose  \
	--ignore-replicated \
	--exclude-unchanged 1 \
	--no-holds \
	--destroy-missing 2y \
	--dry-run \
	$TAG $TARGET \
 	$@

```

Both datasets get processed.

I have noticed that zfs-autobackup uses the 'written' parameter of a filesystem to detect modification after the last snapshot. The 'written' parameter of many of the replicated datasets is > 0 (some, however, are 0!) and therefore the replicated datasets get processed.

Example

On the replicated node:
```
zfs get -H -o property,value -p written rpool/data/subvol-129-disk-0
written	188416
```

On the original node:
```
ssh <original_node> zfs get -H -o property,value -p written rpool/data/subvol-129-disk-0
written	4567040
```

I am using pve 8.2.2 and zfs-autobackup v3.2, installed using pipx.

Any idea what might be happening here?

**Comment by psy0rz** [comment]
Whats going on with written being bigger than 0 on the target you mean?

Check out https://github.com/psy0rz/zfs_autobackup/wiki/Mounting

And try the 3.3 beta version.

**Reply by rholighaus** [reply]
_written_ is checked on the source to determine whether it’s a replica or active file system (_excluded-unchanged_). And the replica will always be mounted on Proxmox, that is system behavior and needed in case you migrate a VM/CT.

As a workaround, I am currently working on a script that checks entries in /etc/pve/[lxc|qemu-server] to flag filesystems as active every time before running zfs-autosnapshot.

**Comment by psy0rz** [comment]
Yes i was talking about the backup target.

**Comment by psy0rz** [comment, marked as answer]
The exclude-unchanged was made for proxmox, i never found out the reason why it changes around 188kb or something.

So just use a value that is big enough.

Indeed your method would be better but more complicated.

**Reply by rholighaus** [reply]
Thanks.
